### PR TITLE
vscode-extensions.chenglou92.rescript-vscode: init at 1.1.3

### PR DESCRIPTION
**Pull request by dlip**
###### Motivation for this change

This vscode plugin had an error running it's analysis binaries on nix

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/#sec-conf-file))
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/misc/vscode-extensions/rescript/rescript-editor-analysis.nix [hunk 1]
```diff
@@ -0,0 +1,24 @@
+{ stdenv, fetchFromGitHub, pkgs }:
```
Comment (SuperSandro2000) [inline pkgs/misc/vscode-extensions/rescript/rescript-editor-analysis.nix:1]: Replace pkgs with the individual packages.

File: pkgs/misc/vscode-extensions/rescript/default.nix [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+in
+vscode-utils.buildVscodeMarketplaceExtension rec {
+  mktplcRef = {
+    name = "rescript-vscode";
+    publisher = "chenglou92";
+    version = "1.1.3";
+    sha256 = "1c1ipxgm0f0a3vlnhr0v85jr5l3rwpjzh9w8nv2jn5vgvpas0b2a";
+  };
+  postPatch = ''
+    rm -rf ${analysisDir}
```
Comment (SuperSandro2000) [inline pkgs/misc/vscode-extensions/rescript/default.nix:18]:
```suggestion
    rm -r ${analysisDir}
```

File: pkgs/misc/vscode-extensions/rescript/rescript-editor-analysis.nix [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+let
+  src = fetchFromGitHub {
+    name = "rescript-vscode";
```
Comment (SuperSandro2000) [inline pkgs/misc/vscode-extensions/rescript/rescript-editor-analysis.nix:4]:
```suggestion
```

File: pkgs/misc/vscode-extensions/rescript/rescript-editor-analysis.nix [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+    owner = "rescript-lang";
+    repo = "rescript-vscode";
+    rev = "8d0412a72307b220b7f5774e2612760a2d429059";
+    sha256 = "rHQtfuIiEWlSPuZvNpEafsvlXCj2Uv1YRR1IfvKfC2s=";
+  };
+in
+stdenv.mkDerivation {
+  inherit src;
```
Comment (SuperSandro2000) [inline pkgs/misc/vscode-extensions/rescript/rescript-editor-analysis.nix:12]: Just move the src into here.

File: pkgs/misc/vscode-extensions/rescript/rescript-editor-analysis.nix [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+  name = "rescript-editor-analysis";
+  nativeBuildInputs = with pkgs; [ bash ocaml ];
```
Comment (SuperSandro2000) [inline pkgs/misc/vscode-extensions/rescript/rescript-editor-analysis.nix:14]:
```suggestion
  nativeBuildInputs = [ ocaml ];
```
bash is in the stdenv IIRC.

File: pkgs/misc/vscode-extensions/rescript/rescript-editor-analysis.nix [hunk 6]
```diff
@@ -0,0 +1,24 @@
...
+
+  installPhase = with pkgs; ''
+    set -euo pipefail
+    cd analysis
+    sed -i 's|/bin/bash|${bash}/bin/bash|g' Makefile
+    make
```
Comment (SuperSandro2000) [inline pkgs/misc/vscode-extensions/rescript/rescript-editor-analysis.nix:20]: Is this compiling anything?

Comment (dlip) [reply pkgs/misc/vscode-extensions/rescript/rescript-editor-analysis.nix:20]: Yes, it is building the analysis binaries. I created an installPhase to be clearer

**Comment by dlip** [comment]
@SuperSandro2000 I've implemented your suggestions, thanks for the review

File: pkgs/misc/vscode-extensions/rescript/rescript-editor-analysis.nix (hunk 6 above)
Comment (SuperSandro2000) [reply pkgs/misc/vscode-extensions/rescript/rescript-editor-analysis.nix:15]:
```suggestion
    make -j$NIX_BUILD_CORES
```

or we move the other two commands to preBuild and let nix handle make correctly.

File: pkgs/misc/vscode-extensions/rescript/rescript-editor-analysis.nix [hunk 7]
```diff
@@ -0,0 +1,22 @@
+{ stdenv, fetchFromGitHub, bash, ocaml }:
+stdenv.mkDerivation {
```
Comment (SuperSandro2000) [inline pkgs/misc/vscode-extensions/rescript/rescript-editor-analysis.nix:2]:
```suggestion
{ stdenv, fetchFromGitHub, bash, ocaml }:

stdenv.mkDerivation {
```

File: pkgs/misc/vscode-extensions/rescript/rescript-editor-analysis.nix [hunk 8]
```diff
@@ -0,0 +1,22 @@
...
+  name = "rescript-editor-analysis";
+  src = fetchFromGitHub {
+    owner = "rescript-lang";
+    repo = "rescript-vscode";
+    rev = "8d0412a72307b220b7f5774e2612760a2d429059";
+    sha256 = "rHQtfuIiEWlSPuZvNpEafsvlXCj2Uv1YRR1IfvKfC2s=";
+  };
+  nativeBuildInputs = [ ocaml ];
```
Comment (SuperSandro2000) [inline pkgs/misc/vscode-extensions/rescript/rescript-editor-analysis.nix:10]:
```suggestion
  };

  nativeBuildInputs = [ ocaml ];
```

File: pkgs/misc/vscode-extensions/rescript/rescript-editor-analysis.nix [hunk 9]
```diff
@@ -0,0 +1,22 @@
+{ stdenv, fetchFromGitHub, bash, ocaml }:
+stdenv.mkDerivation {
+  name = "rescript-editor-analysis";
```
Comment (SuperSandro2000) [inline pkgs/misc/vscode-extensions/rescript/rescript-editor-analysis.nix:3]: New packages should almost always use pname+version.

File: pkgs/misc/vscode-extensions/rescript/rescript-editor-analysis.nix [hunk 10]
```diff
@@ -0,0 +1,22 @@
...
+  buildPhase = ''
+    cd analysis
+    sed -i 's|/bin/bash|${bash}/bin/bash|g' Makefile
+    make
+  '';
+
+  installPhase = ''
+    mkdir -p $out/bin
+    cp ./rescript-editor-analysis.exe $out/bin
+  '';
```
Comment (SuperSandro2000) [inline pkgs/misc/vscode-extensions/rescript/rescript-editor-analysis.nix:21]: Missing meta.

File: maintainers/maintainer-list.nix [hunk 11]
```diff
@@ -2832,6 +2832,12 @@
     githubId = 28980797;
     name = "David Leslie";
   };
+  dlip = {
```
[the remark this answers is not in the document]
Comment (dlip) [reply maintainers/maintainer-list.nix:2835]: https://github.com/NixOS/nixpkgs/pull/140683

File: pkgs/misc/vscode-extensions/rescript/rescript-editor-analysis.nix (hunk 6 above)
Comment (dlip) [reply pkgs/misc/vscode-extensions/rescript/rescript-editor-analysis.nix:15]: Moved to preBuild

**Comment by dlip** [comment]
@SuperSandro2000 thanks again for great suggestions, I've updated the pr

**Pull request merged**